File: backend/semgrep_rules/tests/javascript/kisa_sw_10_ldap_injection.js
function unsafe(req, client) {
  const filter = req.query.filter;
  // ruleid: kisa.sw10.javascript.external-input-to-ldap-filter
  client.search("dc=example,dc=com", { filter }, () => {});
}
function safe(client) {
  // ok: kisa.sw10.javascript.external-input-to-ldap-filter
  client.search("dc=example,dc=com", { filter: "(uid=alice)" }, () => {});
}

File: backend/semgrep_rules/tests/java/kisa_sw_10_ldap_injection.java
import javax.naming.directory.DirContext;
import javax.servlet.http.HttpServletRequest;

class kisa_sw_10_ldap_injection {
    Object unsafe(HttpServletRequest req, DirContext ctx) throws Exception {
        String f = req.getParameter("filter");
        // ruleid: kisa.sw10.java.external-input-to-ldap-filter
        return ctx.search("dc=example,dc=com", f, null);
    }
    Object safe(DirContext ctx) throws Exception {
        // ok: kisa.sw10.java.external-input-to-ldap-filter
        return ctx.search("dc=example,dc=com", "(uid=alice)", null);
    }
}
